miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -758,3 +927,110 @@ proc.get_procedure_metadata
     padw movup.4 mem_loadw drop drop swap
     # => [storage_offset, storage_size]
 end
+
+#! Returns the pointer to the next vacant memory slot if the account was not loaded before, and the 
+#! pointer to the account data otherwise.
+#!
+#! Stack:  [foreign_account_id]
+#! Output: [was_loaded, ptr, foreign_account_id]
+#!
+#! Where: 
+#! - foreign_account_id is the ID of the foreign account whose procedure is going to be executed.
+#! - was_loaded is the binary flag indicating whether the foreign account was already loaded to the 
+#!   memory.
+#! - ptr is the memory pointer to the next empty memory slot or the memory pointer to the account 
+#!   data, depending on the value of the was_loaded flag.
+#!
+#! Panics if:
+#! - the ID of the provided foreign account equals zero.
+#! - the maximum allowed number of foreign account to be loaded (64) was exceeded.
+export.get_foreign_account_ptr
+    # check that foreign account id is not equal zero
+    dup neq.0 assert.err=ERR_FOREIGN_ACCOUNT_ID_IS_ZERO
+    # => [foreign_account_id]
+
+    # check that foreign account id is not equal to the native account id
+    dup exec.memory::get_native_account_id neq assert.err=ERR_FOREIGN_ACCOUNT_ID_EQUALS_NATIVE_ACCT_ID
+
+    # get the initial account data pointer
+    exec.memory::get_native_account_data_ptr
+    # => [curr_account_ptr, foreign_account_id]
+
+    # push the flag to enter the loop
+    push.1
+
+    while.true
+        # drop the flag left from the previous loop
+        movup.2 drop
+        # => [curr_account_ptr, foreign_account_id]
+
+        # move the current account pointer to the next account data block
+        exec.memory::get_account_data_length add
+        # => [curr_account_ptr', foreign_account_id]
+
+        # load the first data word at the current account pointer
+        padw dup.4 mem_loadw
+        # => [FIRST_DATA_WORD, curr_account_ptr', foreign_account_id]
+
+        # check whether the first data words is an empty word, preserve the last value in this word 
+        # (account id)
+        dup.3 movdn.4 padw eqw movdn.8 dropw dropw
+        # => [is_empty_word, last_data_value, curr_account_ptr', foreign_account_id]
+
+        # check whether the current id matches the foreign id 
+        swap dup.3 eq
+        # => [is_equal_id, is_empty_word, curr_account_ptr', foreign_account_id]
+
+        # get the sum of the flags to obtain the loop flag
+        dup movdn.4 add
+        # => [flags_sum, curr_account_ptr', foreign_account_id, is_equal_id]
```

Comment:
Should we not use `and not` here instead of `add`? It seems like we want to keep iterating if `is_equal_id` and `is_empty_word` are both 0.